### DISCUSSION TITLE
Add Files List Side Window

### DISCUSSION THREAD

**Discussion by DudeRenderer** [sumatrapdfreader/sumatrapdf#2690] [Ideas]
If multiple files are open, it is difficult to find them by the top file window name. Add Files List side windows like Bookmarks
check the Notepad++
![image](https://user-images.githubusercontent.com/33353769/172395300-f0b0cbe0-dd70-4137-9e7a-1cb2d955de1f.png)

**Comment by kjk** [comment]
It's not exactly the same thing but if you invoke command palette (`Ctrl + K`) you get a list of tabs first thing in the list. You can then use up /down to select, Enter to open the file. And  narrow down by name with search field.

See https://www.sumatrapdfreader.org/docs/Command-Palette

**Reply by GitHubRulesOK** [reply]
@kjk I have a problem with small factor views that cmd list blocks main view so if it was set to right (not over bookmarks/favorites) it could act as an alternate reference listing, however I guess others like emulating browser vertical tabs on the left

**Reply by DudeRenderer** [reply]
Yep.  I usually open multiply PDF maybe 20+, find specific PDF document is hard. `Ctrl + K` show recently opened document is not list current PDFs within sumatrapdf Editor. 
> vertical tabs on the left nice have~!

**Reply by sumsu** [reply]
Maybe there could be an option to show on top of the command palette current opened tabs, with prefix ie. "*" or "@".

**Reply by GitHubRulesOK** [reply]
@sumsu you can do that by file extension so .p shows .PDFs
![image](https://user-images.githubusercontent.com/12754913/223105386-44219974-ffdc-491c-8cb8-f6625499bf1f.png)

**Reply by sumsu** [reply]
Yes, and no.
Yes, I shows. But it shows all pdf viewed by this instance of sumatra.
My sumatra is running all time for 1-2 weeks, I've got 100-200 viewed pdfs by this time. Some of them with the same name, but different folders/revisions. Part is closed, part still open. So, no, it doesn't show only current open tabs. It is still digging in the haystack.

**Reply by GitHubRulesOK** [reply]
@sumsu The current tabs should be top of list so ignore the 200 below that, however If you need a way to store list of currently opened for program reuse it is about 4 lines of custom script in windows shell see https://github.com/sumatrapdfreader/sumatrapdf/discussions/3045#discussioncomment-3941188

![image](https://user-images.githubusercontent.com/12754913/223117530-8b2b7547-f61b-4c0d-a5b2-32feca0a799b.png)

**Reply by sumsu** [reply]
Thank you for your comprehensive explanation. Now I can see that for closed files there is file path in the round brackets as suffix. This marking as closed tab is better than my idea.

**Comment by DudeRenderer** [comment]
@GitHubRulesOK I'd like to contribute code to the file list feature, with an end result similar to the bookmark, and I'm not sure whether it's likely to be merged into the main branch. 😂😁
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/33353769/42c7ee5a-1ec4-40fa-b77a-52cf4e168b54)

**Reply by DudeRenderer** [reply]
@kjk

**Reply by GitHubRulesOK** [reply]
@kjk seems to have gone aboriginal walkabout last local activity 2 weeks ago https://github.com/kjk/onlinetool.io/commit/5cd8676e2aec86a4e3c0bbea491130a6cb52c4b3 you could try a direct mail but it would be best to show a psudo code of intent not just an illustration which does not show ways to address challenges, file list was attempted as parts of VKBM and BKM so review the methods used there before they were dropped.

**Reply by DudeRenderer** [reply]
Thank you very much for your reply.  I will make a proposal about the filelist. btw, I have a question. What do **VKBM** and **BKM** specifically represent?

**Reply by GitHubRulesOK** [reply]
Sorry that was a typo I meant VBKM added around 2019 was able to hold all filenames initially then with problems changed to pdf only then removed 2021
for some related issues see 
https://github.com/search?q=repo%3Asumatrapdfreader%2Fsumatrapdf+vbkm&type=issues&s=created&o=asc

good illustration of how I was as user finding that helpful for multiple document traversal https://github.com/sumatrapdfreader/sumatrapdf/issues/1357#issue-543903043

Overall a Good Idea just hard to implement and maintain as a fully fledged doc manager.

**Comment by kjk** [comment]
@DudeRenderer adding another permanent panel (like Bookmarks) is quite a bit of work and I don't think it's the best UI.

I would much prefer if command palette (`Ctrl + K`) was improved for this use case.

We already show tabs at the top and you can easily narrow down the file.

Some possible improvements:
* a mode that only shows list of tabs e.g. invoked when you type `@` in search box, similar to how `>` only lists commands, without files
* add `CmdSwitchTab` or `CmdGoToTab` that invokes that mode. Bind it to some key (e.g. `Ctrl + T`) similar to how `Ctrl + Shift + K` invokes `>` mode
* in the list indicate currently active tab (e,g, different color or background)
* show tab number on the right (requires custom painting list control)
* also allow selecting file by typing tab number
* something else?

If you're interested in taking on that, please open a tracking issue and I'm very likely to merge such improvements.

**Reply by DudeRenderer** [reply]
> a mode that only shows list of tabs e.g. invoked when you type @ in search box, similar to how > only lists commands, without files

i prefer it. press specific character, show current window opening tabs.

**Reply by dylanliuh2o** [reply]
Vertical tabs like Microsoft Edge might be a good idea? I usually open more than 10 tabs at the same time, which makes it difficult to find a tab. Vertical tabs may be practical.

<img width="1280" alt="Snipaste_2023-09-16_02-43-09" src="https://github.com/sumatrapdfreader/sumatrapdf/assets/44540113/8738b92b-f08d-4cd3-a088-845da85327dc">
<img width="1280" alt="Snipaste_2023-09-16_02-45-03" src="https://github.com/sumatrapdfreader/sumatrapdf/assets/44540113/6a5527af-65a1-493c-8871-43df98e74512">
<img width="1280" alt="Snipaste_2023-09-16_02-44-47" src="https://github.com/sumatrapdfreader/sumatrapdf/assets/44540113/e7d61f8e-f581-4881-8916-c2463505567a">

**Reply by foooal** [reply]
yes, especially for a file with long name

**Reply by Jammerlee** [reply]
A side window like this would be super helpful. Especially if I can switch it between views for chapters and/or pages

**Comment by kjk** [comment]
I've opened https://github.com/sumatrapdfreader/sumatrapdf/issues/3659 to track Command Palette Improvements for this.

Already implemented `@` to only show tabs

**Comment by DudeRenderer** [comment]
I have fixed some bugs that onlyTabs in PR #3662

**Reply by kjk** [reply]
Thanks, I appreciate that.

**Comment by shallow-beach** [comment]
inspiration could be taken from Chrome's 'Add tab to group' feature. it is a clean intermediary when working in the [10,100] range of tabs, although of course would need to be adapted to Sumatra minimality - and its usefulness is pretty limited to that range

**Comment by kjk** [comment]
You can now (latest 3.5 pre-release) use  `Alt + K` to get a list of all tabs and switch between them.
